Sources/SpelledPitch/PitchSpeller/Wetherfield/SpellingInverter.swift

```diff
@@ -0,0 +1,423 @@
+//
+//  SpellingInverter.swift
+//  SpelledPitch
+//
+//  Created by Benjamin Wetherfield on 2/14/19.
+//
+
+import DataStructures
+import Pitch
+
+struct SpellingInverter {
+    
+    // MARK: - Instance Properties
+    
+    // The pre-assigned `DirectedGraph` which stands in for the `FlowNetwork` that
+    // would solve to the set of spellings passed into the `SpellingInverter`.
+    var flowNetwork: DirectedGraph<PitchSpeller.AssignedNode>
+    
+    let pitchClass: (Int) -> Pitch.Class?
+}
+
+extension SpellingInverter {
+    
+    // MARK: - Type Aliases
+    
+    typealias AssignedEdge = OrderedPair<PitchSpeller.AssignedNode>
+    typealias UnassignedEdge = OrderedPair<PitchSpeller.UnassignedNode>
+    typealias PitchedEdge = UnorderedPair<FlowNode<Cross<Pitch.Class,Tendency>>>
+}
+
+extension SpellingInverter {
+    
+    // MARK: - Initializers
+    
+    init(spellings: [Int: Pitch.Spelling], parsimonyPivot: Pitch.Spelling = .init(.d)) {
+        self.flowNetwork = DirectedGraph(internalNodes: internalNodes(spellings: spellings))
+        self.pitchClass = { int in spellings[int]?.pitchClass }
+        
+        let specificEdgeScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> = (upDownEdgeScheme.pullback(nodeMapper) + sameEdgeScheme.pullback(nodeMapper))
+            * connectDifferentInts
+        
+        let sameIntEdgesScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> =
+            sameIntsScheme * connectSameInts
+        
+        let specificSourceScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> =
+            sourceEdgeLookupScheme.pullback(nodeMapper)
+        
+        let specificSinkScheme: DirectedGraphScheme<PitchSpeller.UnassignedNode> =
+            sinkEdgeLookupScheme.pullback(nodeMapper)
+        
+        let maskScheme: DirectedGraphScheme<PitchSpeller.AssignedNode> =
```

Comment:
For clarity, I might just split this out into two expressions:

```Swift
        let allSchemes: DirectedGraphScheme<PitchSpeller.AssignedNode> = [
                specificEdgeScheme,
                sameIntEdgesScheme,
                specificSourceScheme,
                specificSinkScheme
            ]
        let maskScheme = allSchemes
            .reduce(DirectedGraphScheme { _ in false }, +)
            .pullback({ $0.unassigned })
```

Reply:
perfect